app/business_model/helpers.py

```diff
@@ -101,3 +102,10 @@ def available_models(score, grid_condition):
                 if "cooperative" not in B_MODELS[k]["Name"]:
                     models.append((k, B_MODELS[k]["Verbose"]))
     return models
+
+
+def validate_percent(value):
+    if 0 <= value <= 100:
+        pass
+    else:
+        raise ValidationError(_("Please provide a value between 0 and 100"))
```

Comment:
This doesn't really do much because if the user provides a value between 0 and 1 as a percentage it will still pass the validation, even though the percentage would be in the wrong format. Maybe the only way to properly validate percentages it to actually have them be input as numbers between 0 and 1, even if it is a slightly more confusing input for non-experts.

Reply:
I also find factors should be float between 0 and 1, but it was a specific requests to be able to input numbers between 0 and 100. I guess most of the user will input numbers between 0 and 100 and if they input, say, 0.5, then they really mean 0.5%, ie 0.005. So I don't really see a problem here